openedx/core/djangoapps/content/learning_sequences/api/processors/cohort_partition_groups.py

```diff
@@ -0,0 +1,87 @@
+# lint-amnesty, pylint: disable=missing-module-docstring
+import logging
+from datetime import datetime
+
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.core import types
+from openedx.core.djangoapps.course_groups.cohorts import (
+    get_cohort,
+    get_cohorted_user_partition_id,
+    get_group_info_for_cohort,
+)
+
+from .base import OutlineProcessor
+
+log = logging.getLogger(__name__)
+
+
+class CohortPartitionGroupsOutlineProcessor(OutlineProcessor):
+    """
+    Processor for applying cohort user partition groups.
+
+    """
+    def __init__(self, course_key: CourseKey, user: types.User, at_time: datetime):
+        super().__init__(course_key, user, at_time)
```

Comment:
We can avoid importing `CourseKey` and `types.User` by doing something like:
```suggestion
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
```

Reply:
I see what you mean but I tried to follow what was done in other processors : [EnrollmentTrackPartitionGroupsOutlineProcessor](https://github.com/openedx/edx-platform/blob/b97007e1823320c3d97b5a84b0c988d888250e84/openedx/core/djangoapps/content/learning_sequences/api/processors/enrollment_track_partition_groups.py#L31), [ContentGatingOutlineProcessor](https://github.com/openedx/edx-platform/blob/b97007e1823320c3d97b5a84b0c988d888250e84/openedx/core/djangoapps/content/learning_sequences/api/processors/content_gating.py#L25), [ScheduleOutlineProcessor](https://github.com/openedx/edx-platform/blob/b97007e1823320c3d97b5a84b0c988d888250e84/openedx/core/djangoapps/content/learning_sequences/api/processors/schedule.py#L37-L38).

Reply:
I think either way is fine, FWIW. Unless it's causing some kind of error in the linting?

Reply:
It is not causing any errors.